fields/types/list/ListField.js

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable react/jsx-no-bind */
+
+import assign from 'object-assign';
+import React from 'react';
+import Field from '../Field';
+import Domify from 'react-domify';
+
+import { Fields } from 'FieldTypes';
+import { Button, GlyphButton } from '../../../admin/client/App/elemental';
+import InvalidFieldType from '../../../admin/client/App/shared/InvalidFieldType';
+
+function generateId () {
+	return Math.random().toString(36).substr(2, 9);
+};
+
+module.exports = Field.create({
+	displayName: 'ListField',
+	statics: {
+		type: 'List',
+	},
+	propTypes: {
+		fields: React.PropTypes.object.isRequired,
+		label: React.PropTypes.string,
+		onChange: React.PropTypes.func.isRequired,
+		path: React.PropTypes.string.isRequired,
+		value: React.PropTypes.array,
+	},
+	addItem () {
+		const value = this.props.value.slice();
+		value.push({
+			_id: generateId(),
+			_isNew: true,
+		});
+		this.props.onChange({
+			path: this.props.path,
+			value: value,
+		});
+	},
+	removeItem (index) {
+		const oldValue = this.props.value;
+		const value = oldValue.slice(0, index).concat(oldValue.slice(index + 1));
+		this.props.onChange({
+			path: this.props.path,
+			value: value,
+		});
+	},
+	handleFieldChange (index, event) {
+		const value = this.props.value.slice();
+		const update = value[index];
+		update[event.path] = event.value;
```

Comment:
* Are you assuming here that the list schema is only one level deep? Don't we want arbitrarily nested objects?
* You are mutating that object

I think the field code could become much cleaner by keeping the editor object in Redux and sending changes as `(pathArray, changeDescription)` tuples. That way, the object editing code is only in one place, and the changes don't need to trickle up.

This list element would not be responsible for patching object changes into itself, and different list implementations would all be able to use that code.

Actually implementing this won't even be that much code, the net effect will probably be less code.